src/CompanyService.Broker/Consumers/GetDepartmentsConsumer.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LT.DigitalOffice.CompanyService.Data.Interfaces;
+using LT.DigitalOffice.CompanyService.Models.Db;
+using LT.DigitalOffice.CompanyService.Models.Dto.Enums;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.Models.Broker.Models.Company;
+using LT.DigitalOffice.Models.Broker.Requests.Company;
+using LT.DigitalOffice.Models.Broker.Responses.Company;
+using MassTransit;
+
+namespace LT.DigitalOffice.CompanyService.Broker.Consumers
+{
+  public class GetDepartmentsConsumer : IConsumer<IGetDepartmentsRequest>
+  {
+    private readonly IDepartmentRepository _repository;
+
+    private object GetDepartment(IGetDepartmentsRequest request)
+    {
+      List<DbDepartment> dbDepartments = new();
+
+      if (request.DepartmentsIds != null && request.DepartmentsIds.Any())
```

Comment:
you can paste request.DepartmentsIds to some variable to not write request.DepartmentsIds 3rd time + it will faster

Reply:
why is it faster

Reply:
cause when you try to get some property or method via '.' in fact call 
access operator (method). and to not cal it func it every time you can set result to variable and use it. in c++ you can overload it (https://docs.microsoft.com/ru-ru/cpp/cpp/member-access?view=msvc-160)

Reply:
fixed